aws/resource_aws_api_gateway_integration.go

```diff
@@ -55,6 +55,21 @@ func resourceAwsApiGatewayIntegration() *schema.Resource {
 				}, false),
 			},
 
+			"connection_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  apigateway.ConnectionTypeInternet,
+				ValidateFunc: validation.StringInSlice([]string{
+					apigateway.ConnectionTypeInternet,
+					apigateway.ConnectionTypeVpcLink,
+				}, false),
+			},
+
+			"connection_id": {
```

Comment:
We're missing `d.Set("connection_id", integration.ConnectionId)` in the read function 😄